src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -0,0 +1,84 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkBase;
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj2.command.*;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+import frc.lib.Neo;
+
+import java.util.function.DoubleSupplier;
+
+import static frc.robot.Constants.IntakeConstants.*;
+
+public class Intake extends SubsystemBase {
+    public int targetPos = INTAKE_ANGLE.IDLE.angle;
+
+    private final PIDController anglePIDcontroller = new PIDController(PIDGains.kp, PIDGains.ki, PIDGains.kd);
+    private final ArmFeedforward angleFFcontroller = new ArmFeedforward(FFangleGains.ks, FFangleGains.kg, FFangleGains.kv);
+
+    private final Neo intakeMotor = new Neo(0);
+    private final Neo angleMotor = new Neo(0);
+
+    private final DutyCycleEncoder intakeEncoder = new DutyCycleEncoder(0);
+
+    private final DigitalInput beamBreak = new DigitalInput(0);
+    private final Trigger beamBreakTrigger = new Trigger(beamBreak::get);
+
+    public Intake(){
+        intakeMotor.setIdleMode(CANSparkBase.IdleMode.kBrake);
+        intakeMotor.setInverted(false);
+        intakeMotor.setConversionFactors(0,0);
+
+        angleMotor.setInverted(false);
+        angleMotor.setIdleMode(CANSparkBase.IdleMode.kBrake);
+
+        intakeEncoder.setDistancePerRotation(360);
+        intakeEncoder.setPositionOffset(0);
+    }
+
+    public Command setRollerSpeedCommand(DoubleSupplier speed){
+        return new RunCommand(()-> intakeMotor.set(speed.getAsDouble())).until(beamBreakTrigger);
+    }
```

Comment:
don't put the end condition here, it's confusing, does this command set the intake speed, or does it stop it? or both?
each command does one thing, this command sets a speed to the intake motor...
the end condition will come at the ```intakeCommand```